auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailActivity.java

```diff
@@ -94,21 +106,45 @@ protected void onCreate(Bundle savedInstanceState) {
 
         togglePasswordImage.setOnClickListener(new PasswordToggler(mPasswordEditText));
 
-        mNameEditText = (EditText) findViewById(R.id.name);
 
-        mPasswordFieldValidator = new PasswordFieldValidator((TextInputLayout)
-                findViewById(R.id.password_layout),
+        mPasswordFieldValidator = new PasswordFieldValidator(
+                (TextInputLayout) findViewById(R.id.password_layout),
                 getResources().getInteger(R.integer.min_password_length));
         mNameValidator = new RequiredFieldValidator((TextInputLayout) findViewById(R.id.name_layout));
         mEmailFieldValidator = new EmailFieldValidator((TextInputLayout) findViewById(R.id.email_layout));
 
-        if (email != null) {
+        mNameEditText = (EditText) findViewById(R.id.name);
+        mEmailEditText = (EditText) findViewById(R.id.email);
+
+        mEmailEditText.setOnFocusChangeListener(this);
+        mNameEditText.setOnFocusChangeListener(this);
+        mPasswordEditText.setOnFocusChangeListener(this);
+        findViewById(R.id.button_create).setOnClickListener(this);
+
+        setUpTermsOfService();
+
+        // Activity rotated
+        if (savedInstanceState != null) {
+            mPrevEmail = savedInstanceState.getString(PREV_EMAIL);
+            return;
+        }
+
+        String email = getIntent().getStringExtra(ExtraConstants.EXTRA_EMAIL);
+        if (!TextUtils.isEmpty(email)) {
             mEmailEditText.setText(email);
-            mEmailEditText.setEnabled(false);
+            mNameEditText.requestFocus();
+            return;
+        }
+
+        if (mActivityHelper.getFlowParams().smartLockEnabled) {
+            showEmailAutoCompleteHint();
```

Comment:
Why are we doing this only if smart lock is enabled? I think users should get autocomplete whether or not devs have an aversion to smart lock.